engine/src/main/java/org/camunda/bpm/engine/impl/migration/instance/MigratingProcessInstance.java

```diff
@@ -119,37 +121,64 @@ public static MigratingProcessInstance initializeFrom(CommandContext commandCont
     Map<String, List<MigrationInstruction>> organizedInstructions = organizeInstructionsBySourceScope(migrationPlan);
 
     for (ActivityInstance instance : activityInstances) {
-      ActivityImpl sourceActivity = sourceProcessDefinition.findActivity(instance.getActivityId());
-
-      List<MigrationInstruction> instructionCandidates = organizedInstructions.get(sourceActivity.getId());
-      MigrationInstruction applyingInstruction = null;
-      ActivityImpl targetActivity = null;
-
-      if (instructionCandidates != null && instructionCandidates.size() > 0) {
-        // TODO: this could be more than one when we support conditional instructions
-        applyingInstruction = instructionCandidates.get(0);
-        targetActivity = targetProcessDefinition.findActivity(applyingInstruction.getTargetActivityIds().get(0));
-
-      }
-      else {
-        if (instance.getChildActivityInstances().length == 0) {
-          unmappedLeafInstances.add(instance);
-        }
-      }
-
-      MigratingActivityInstance migratingInstance = migratingProcessInstance.addActivityInstance(
-        applyingInstruction,
-        instance,
-        sourceActivity,
-        targetActivity,
-        mapping.getExecution(instance));
-
-      if (sourceActivity.getActivityBehavior() instanceof UserTaskActivityBehavior) {
-        List<TaskEntity> tasks = migratingInstance.representativeExecution.getTasks();
-        migratingInstance.addDependentInstance(new MigratingTaskInstance(tasks.get(0), migratingInstance));
-      }
-    }
-
+		ActivityImpl sourceActivity = sourceProcessDefinition.findActivity(instance.getActivityId());
+
+		List<MigrationInstruction> instructionCandidates = organizedInstructions.get(sourceActivity.getId());
+		MigrationInstruction applyingInstruction = null;
+		ActivityImpl targetActivity = null;
+
+		if (instructionCandidates != null && instructionCandidates.size() > 0) {
+			
+			int instructionCandidatesSize = instructionCandidates.size(); 
+			boolean conditionSatisfied = false;
+			if( instructionCandidatesSize > 1)
+			{
+				for(int index =0; index <instructionCandidates.size();index++)
+				{
+					applyingInstruction = instructionCandidates.get(index);
+					if(applyingInstruction == null)//java.util.List allows null object as well
+						continue;
+					Condition condition = applyingInstruction.getCondition();
+					//TODO: Retrieving the delegate event has to be checked for mapping condition
+					if(condition != null && condition.shouldMap(DelegateEvent.getEvents().get(0)))
```

Comment:
Thank you so much for your review. Ya it should be as you have mentioned and  that was the doubt i was having earlier regarding retrieval of DelegateExecution from instance.

Kindly let me know whether i can make the respective change and commit it to the branch i created (conditionalMigration)